cmake/dependencies/pyAMReX.cmake

```diff
@@ -74,7 +74,7 @@ option(ImpactX_pyamrex_internal "Download & build pyAMReX" ON)
 set(ImpactX_pyamrex_repo "https://github.com/AMReX-Codes/pyamrex.git"
     CACHE STRING
     "Repository URI to pull and build pyamrex from if(ImpactX_pyamrex_internal)")
-set(ImpactX_pyamrex_branch "25.07"
+set(ImpactX_pyamrex_branch "f884dda4beebdd955aaba8f82a8a9530b8185bb5"
```

Comment:
See https://github.com/AMReX-Codes/pyamrex/commit/f884dda4beebdd955aaba8f82a8a9530b8185bb5 and https://github.com/AMReX-Codes/pyamrex/commit/8b1407e8fc5d7259107d2f81dd5e20147a31b62b.